mongo/integration/client_test.go

```diff
@@ -526,11 +526,11 @@ func TestClient(t *testing.T) {
 
 		// Assert that the minimum RTT is eventually >250ms.
 		topo := getTopologyFromClient(mt.Client)
-		assert.Soon(mt, func(ctx context.Context) {
+		callback := func() {
 			for {
 				// Stop loop if callback has been canceled.
 				select {
-				case <-ctx.Done():
+				case <-context.Background().Done():
```

Comment:
`context.Background()` constructs a new empty context. `Done()` returns a channel that is closed when the context is canceled. Given that, this block will never execute. 

See here for a code explanation: https://go.dev/play/p/OuqaCCQKfgD
See here for a valid use case: https://go.dev/play/p/L6JmR_D2G0a

Suggest removing the entire select block, as it does nothing.